Manager/Manager.php

```diff
@@ -78,6 +89,32 @@ public function save($entity, array $options = array())
      */
     public function remove($entity)
     {
-        $this->remover->remove($entity);
+        if (!$this->isLinkedToProduct($entity)) {
+            $this->remover->remove($entity);
+        }
+    }
+
+    /**
+     * Check if the entity is linked to one or more products
+     *
+     * @param ReferenceDataInterface|object $entity
+     *
+     * @return bool
+     */
+    protected function isLinkedToProduct($entity)
+    {
+        $metadata = $this->em->getClassMetadata(Attribute::class);
+        $repository = new AttributeRepository($this->em, $metadata);
```

Comment:
Create a service

Reply:
Yep, but i was feeling like missing something. If not other solution, i'll create the service.